### PR TITLE
[flow] Move CloneProducersIntoDispatchRegions into its own pass

### PR DESCRIPTION
**Pull request by antiagainst**
By cloning producers into dispatch regions in its own pass before the main FormDispatchWorkgroupsPass, we can enable running additional transformations afterwards that would need the full dispatch content but don't want to handle explicit captures as materialized as dispatch workgroup operands and block arguments.

This forms a nice structure together with FormDispatchRegionsPass, where we only moves the root compute op into the dispatch region, so that we can run additional transformations afterwards with a simple region and without bothering producers.

Progress towards https://github.com/openxla/iree/issues/12230

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.td [hunk 1]
```diff
@@ -84,6 +84,13 @@ def FormDispatchWorkgroups :
   ];
 }
 
+def CloneProducersIntoDispatchRegions :
+    InterfacePass<"iree-flow-clone-producers-into-dispatch-regions", "mlir::FunctionOpInterface"> {
+  let summary = "Clone producers into dispatch regions to be isolated above";
+  let constructor = "mlir::iree_compiler::IREE::Flow::createCloneProducersIntoDispatchRegionsPass()";
+}
+
+
```
Comment (benvanik) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.td]: extra line

**Review by benvanik — APPROVED** [review]
(lgtm but mahesh may want to look before merging!)

**Comment by MaheshRavishankar** [comment]
Looks good to me. Unfortunately we don't have good tracking of dispatch region count in CI yet (we have been doing this manually). @pzread can you help make sure this doesn't cause regressions.

**Comment by pzread** [comment]
> Looks good to me. Unfortunately we don't have good tracking of dispatch region count in CI yet (we have been doing this manually). @pzread can you help make sure this doesn't cause regressions.

Will do later today and post results here

**Comment by pzread** [comment]
There is no change in the number of dispatches before and after this patch (measured from `--iree-scheduling-dump-statistics-format=json`) on x86_64 benchmarks

**Review by MaheshRavishankar — APPROVED** [review]

**Comment by antiagainst** [comment]
Thanks @pzread!

**Pull request merged**